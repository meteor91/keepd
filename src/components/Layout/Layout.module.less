@import "styles/colors.less";

.layout {
  display: flex;
}

.sidebar {
  width: 270px;
  height: 100vh;
  background: @blueGray;
  padding: 10px;
}

.content {
  padding: 10px;
}