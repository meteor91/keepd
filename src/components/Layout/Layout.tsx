import React from 'react';

import styles from './Layout.module.less';

interface SidebarProps {
    children?: React.ReactNode;
}
const Sidebar: React.FC<SidebarProps> = ({ children }) => {
    return (
        <div className={styles.sidebar}>
            {children}
        </div>
    );
};

interface ContentProps {
    children?: React.ReactNode;
}
const Content: React.FC<ContentProps> = ({ children }) => {
    return (
        <div className={styles.content}>
            {children}
        </div>
    );
};

interface LayoutProps {
    children?: React.ReactNode;
}
export const Layout = ({ children }: LayoutProps): React.ReactElement => {
    return (
        <div className={styles.layout}>
            {children}
        </div>
    );
};

Layout.Sidebar = Sidebar;
Layout.Content = Content;
