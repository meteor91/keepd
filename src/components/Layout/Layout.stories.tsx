import React from 'react';
import type { Meta, StoryObj } from '@storybook/react';

import { Layout } from './Layout';
import { NavBar } from '../NavBar';
import { DeviceTabletIcon } from '../../internal/svg-icons/DeviceTabletIcon';
import { PackageIcon } from '../../internal/svg-icons/PackageIcon';

const meta: Meta<typeof Layout> = {
    title: 'Layout',
    component: Layout,
};

export default meta;
type Story = StoryObj<typeof Layout>;

export const Basic: Story = {
    render: () => (
        <Layout>
            <Layout.Sidebar>
                <NavBar items={items}/>
            </Layout.Sidebar>
            <Layout.Content>
                some content
            </Layout.Content>
        </Layout>
    ),
};

const items = [
    {
        icon: <DeviceTabletIcon />,
        label: 'item 1',
    },
    {
        icon: <PackageIcon />,
        label: 'item 2',
    },
];
