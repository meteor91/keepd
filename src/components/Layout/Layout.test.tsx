import React from 'react';
import { render } from '@testing-library/react';

import { Layout } from './Layout';

describe('Layout', () => {
    it('should render correctly', () => {
        const { container } = render(
            <Layout>
                <Layout.Sidebar>sidebar</Layout.Sidebar>
                <Layout.Content>content</Layout.Content>
            </Layout>,
        );
        expect(container.firstChild).toMatchSnapshot();
    });
});
