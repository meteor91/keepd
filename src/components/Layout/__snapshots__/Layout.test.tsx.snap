// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout should render correctly 1`] = `
<div
  class="layout"
>
  <div
    class="sidebar"
  >
    sidebar
  </div>
  <div
    class="content"
  >
    content
  </div>
</div>
`;
